x/emissions/keeper/actor_utils/worker.go

```diff
@@ -241,66 +200,22 @@ func insertForecastsFromTopForecasters(
 		/// Now do filters on each forecaster
 		// Ensure that we only have one forecast per forecaster. If not, we just take the first one
 		if _, ok := forecastsByForecaster[forecast.Forecaster]; !ok {
-			// Check if the forecaster is registered
-			isForecasterRegistered, err := k.IsWorkerRegisteredInTopic(ctx, topicId, forecast.Forecaster)
-			if err != nil {
-				ctx.Logger().Warn("Error checking forecaster registration: ", topicId, ", nonce: ", nonce, "for forecaster: ", forecast.Forecaster)
-				continue
-			}
-			if !isForecasterRegistered {
-				ctx.Logger().Warn("Forecaster not registered, topic: ", topicId, ", nonce: ", nonce, "for forecaster: ", forecast.Forecaster)
-				continue
-			}
-
-			// Examine forecast elements to verify that they're for inferers in the current set.
-			// We assume that set of inferers has been verified above.
-			// We keep what we can, ignoring the forecaster and their contribution (forecast) entirely
-			// if they're left with no valid forecast elements.
-			acceptedForecastElements := make([]*types.ForecastElement, 0)
-			for _, el := range forecast.ForecastElements {
-				if _, ok := acceptedInferersOfBatch[el.Inferer]; ok {
-					acceptedForecastElements = append(acceptedForecastElements, el)
-				}
-			}
-
-			// Discard if empty
-			if len(acceptedForecastElements) == 0 {
-				continue
-			}
-
-			/// Filtering done now, now write what we must for inclusion
-
-			// Get the latest score for each forecaster => only take top few by score descending
-			latestScore, err := k.GetLatestForecasterScore(ctx, topicId, forecast.Forecaster)
-			if err != nil {
-				continue
-			}
-			latestForecasterScores[forecast.Forecaster] = latestScore
+			latestForecaster = append(latestForecaster, forecast)
 			forecastsByForecaster[forecast.Forecaster] = forecast
 		}
 	}
 
-	/// If we pseudo-random sample from the non-sybil set of reputers, we would do it here
-	topForecasters := FindTopNByScoreDesc(maxTopWorkersToReward, latestForecasterScores, nonce.BlockHeight)
-
-	// Build list of forecasts that pass all filters
-	// AND are from top performing forecasters among those who have submitted forecasts in this batch
-	forecastsFromTopForecasters := make([]*types.Forecast, 0)
-	for _, worker := range topForecasters {
-		forecastsFromTopForecasters = append(forecastsFromTopForecasters, forecastsByForecaster[worker])
-	}
-
 	// Though less than ideal because it produces less-acurate network inferences,
 	// it is fine if no forecasts are accepted
 	// => no need to check len(forecastsFromTopForecasters) == 0
 
 	// Ensure deterministic ordering
-	sort.Slice(forecastsFromTopForecasters, func(i, j int) bool {
-		return forecastsFromTopForecasters[i].Forecaster < forecastsFromTopForecasters[j].Forecaster
+	sort.Slice(latestForecaster, func(i, j int) bool {
+		return latestForecaster[i].Forecaster < latestForecaster[j].Forecaster
```

Comment:
int his "1-tier", online design, we should only be adding values from actors of better previous score, meaning we "skim the top as we go", with each successive data bundle. this means that we may not need to sort at all, and we certainly shouldn't need to sort when we close the nonce